src/internal/kopia/snapshot_manager.go

```diff
@@ -275,59 +271,54 @@ func fetchPrevManifests(
 func fetchPrevSnapshotManifests(
 	ctx context.Context,
 	sm snapshotManager,
-	oc *OwnersCats,
+	reasons []Reason,
 	tags map[string]string,
 ) []*ManifestEntry {
-	if oc == nil {
-		return nil
-	}
-
 	mans := map[manifest.ID]*ManifestEntry{}
 
 	// For each serviceCat/resource owner pair that we will be backing up, see if
 	// there's a previous incomplete snapshot and/or a previous complete snapshot
 	// we can pass in. Can be expanded to return more than the most recent
 	// snapshots, but may require more memory at runtime.
-	for _, serviceCat := range oc.ServiceCats {
-		for resourceOwner := range oc.ResourceOwners {
-			found, err := fetchPrevManifests(
-				ctx,
-				sm,
-				mans,
-				serviceCat,
-				resourceOwner,
-				tags,
+	for _, reason := range reasons {
+		found, err := fetchPrevManifests(
+			ctx,
+			sm,
+			mans,
+			reason,
+			tags,
+		)
+		if err != nil {
+			logger.Ctx(ctx).Warnw(
+				"fetching previous snapshot manifests for service/category/resource owner",
+				"error",
+				err,
+				"service",
+				reason.Service.String(),
+				"category",
+				reason.Category.String(),
 			)
-			if err != nil {
-				logger.Ctx(ctx).Warnw(
-					"fetching previous snapshot manifests for service/category/resource owner",
-					"error",
-					err,
-					"service/category",
-					serviceCat,
-				)
-
-				// Snapshot can still complete fine, just not as efficient.
+
+			// Snapshot can still complete fine, just not as efficient.
+			continue
+		}
+
+		// If we found more recent snapshots then add them.
+		for _, m := range found {
+			found := mans[m.ID]
```

Comment:
I know this is movement of old code, but it just feels weird to re-use the ranged variable here.
```suggestion
		for _, m := range found {
			man := mans[m.ID]
```